backpack/extensions/firstorder/variance/variance_base.py

```diff
@@ -1,30 +1,89 @@
+"""Contains VarianceBaseModule."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, List, Tuple
+
+from torch import Tensor
+from torch.nn import Module
+
 from backpack.extensions.firstorder.base import FirstOrderModuleExtension
 
+if TYPE_CHECKING:
+    from backpack.extensions import Variance
+    from backpack.extensions.firstorder.gradient.base import GradBaseModule
+    from backpack.extensions.firstorder.sum_grad_squared.sgs_base import SGSBase
+
 
 class VarianceBaseModule(FirstOrderModuleExtension):
-    def __init__(self, params, grad_extension, sgs_extension):
-        super().__init__(params=params)
-        self.grad_ext = grad_extension
-        self.sgs_ext = sgs_extension
+    """Base class for extensions calculating variance."""
+
+    def __init__(
+        self,
+        params: List[str],
+        grad_extension: GradBaseModule,
+        sgs_extension: SGSBase,
+    ):
+        """Initialization.
+
+        Creates a function named after each parameter.
+
+        Args:
+            params: list of parameter names
+            grad_extension: the extension calculating grad.
+            sgs_extension: the extension calculating squared_grad_sum.
+        """
+        self.grad_ext: GradBaseModule = grad_extension
+        self.sgs_ext: SGSBase = sgs_extension
```

Comment:
@schaefertim Are these type annotations necessary? (Aren't they implied by the types specified in the constructor?)